local/compose/create.go

```diff
@@ -595,9 +595,19 @@ func (s *composeService) buildContainerVolumes(ctx context.Context, p types.Proj
 		}
 	}
 	for _, m := range mountOptions {
-		if m.Type == mount.TypeBind || m.Type == mount.TypeTmpfs {
-			mounts = append(mounts, m)
+		if m.Type != mount.TypeBind && m.Type != mount.TypeTmpfs {
+			continue
+		}
+		if m.Type == mount.TypeBind {
+			// mkdir if bind source path does not exist
+			if _, err := os.Stat(m.Source); err != nil && os.IsNotExist(err) {
+				err = os.Mkdir(m.Source, 0755)
```

Comment:
👎 doing so you create source on the CLI host, which might not be the dockerHost. Obviously this is what most people use for development, but this is wrong here. Need to understand how our bind configuration differs from docker/cli doing the same